### PR TITLE
added getting product ID from url parameter

### PR DESCRIPTION
**Pull request by cmac0351**
@mattsaxe17 @sunikkim 

I had posted in slack and then got a chance to implement it. With this, we can use a product id as a url parameter, but have the first product as default if none is entered.

### CODE REVIEW

**Comment by cmac0351** [comment]
well, I may have got a little carried away on this branch, but I added htmlWebpackPlugin to our config. This makes it so webpack generates index.html when it builds. This is to setup something I want to do with caching. I also added an npm script called build, to just run build on our EC2 instances.

**Comment by cmac0351** [comment]
....and down the rabbit hole I went, so now in addition to generating the index.html file automatically, we now get an md5 hash as part of our css and bundle file.

The idea here is that if the file changes, the hash changes, so there is a different filename. This will keep the browser from using a cached file when there is a change. However, if no changes were made, the filename will match what the browser has cached.

Since index.html is generated on build, it is injected with the correct bundle name with the hash.

File: client/src/index.js [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
-
 import App from './components/App/App.jsx';
 
+const queryString = window.location.search;
+const urlParams = new URLSearchParams(queryString);
+const productId = urlParams.has('product') ? urlParams.get('product') : 36300;
+
 ReactDOM.render(
-  <App />,
+  <App productId={productId}/>,
```
Comment (sunikkim) [inline client/src/index.js:10]: nice!

**Review by sunikkim — APPROVED** [review]
awesome stuff!

File: client/src/template.html [hunk 2]
```diff
@@ -1,10 +1,9 @@
 <!DOCTYPE html>
 <html lang="en">
   <head>
-    <Title>Chevre</Title>
+    <Title>Le Chevre Boutique</Title>
```
Comment (mattsaxe17) [inline client/src/template.html:4]: Love it!😂

**Review by mattsaxe17 — APPROVED** [review]

**Pull request merged**